### PR TITLE
Intro bindingsites

### PR DESCRIPTION
**Pull request by elisagdelope**
## Description

Binding site prediction is a crucial task in computational biology and drug discovery. This tutorial aims to provide users with an introductory background on binding sites and computational methods to study and identify them.

This PR adds a new Jupyter notebook tutorial for binding site prediction using DeepChem. The tutorial covers the following topics:

- Introduction to binding sites: An overview of what binding sites are and their importance.
- Basic concepts: Fundamental concepts related to binding sites.
- Types of binding sites: Different types of binding sites and their characteristics.
- Computational methods to study binding sites: Various computational approaches to analyze binding sites.
- DeepChem tools to study binding sites: Tools and functionalities provided by DeepChem for binding site analysis.
- A case for protein language models: Discussion on the application of protein language models in binding site prediction.
- How does a binding pocket look like?: Visual representation of binding pockets.
- Further reading: Additional resources and references for further study.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by GreatRSingh** [comment]
You should add `Citing This Tutorial` section.

ref - https://github.com/deepchem/deepchem/blob/master/examples/tutorials/Cell_Counting_Tutorial.ipynb

File: examples/tutorials/Introduction_to_Binding_Sites.ipynb [hunk 1]
```diff
@@ -0,0 +1 @@
....41           C  \\nATOM    343  O   LEU A 266      10.793  58.124   0.062  1.00 50.02           O  \\nATOM    344  CB  LEU A 266       8.401  56.228  -0.006  1.00 51.74           C  \\nATOM    345  CG  LEU A 266       8.126  55.926   1.468  1.00 53.35           C  \\nATOM    346  CD1 LEU A 266       6.696  56.323   1.812  1.00 50.66           C  \\nATOM    347  CD2 LEU A 266       8.340  54.444   1.732  1.00 48.73           C  \\nATOM    348  N   THR A 267       9.252  59.247   1.266  1.00 47.07           N  \\nATOM    349  CA  THR A 267      10.247  59.930   2.081  1.00 44.50           C  \\nATOM    350  C   THR A 267      10.635  58.958   3.188  1.00 41.84           C  \\nATOM    351  O   THR A 267       9.771  58.382   3.848  1.00 40.72           O  \\nATOM    352  CB  THR A 267       9.691  61.212   2.708  1.00 45.02           C  \\nATOM    353  OG1 THR A 267       9.328  62.129   1.672  1.00 46.49           O  \\nATOM    354  CG2 THR A 267      10.736  61.856   3.606  1.00 40.98           C  \\nATOM    355  N   VAL A 268      11.935  58.779   3.386  1.00 37.58           N  \\nATOM    356  CA  VAL A 268      12.429  57.849   4.387  1.00 36.97           C  \\nATOM    357  C   V
```
Comment (rbharath) [inline examples/tutorials/Introduction_to_Binding_Sites.ipynb:1]: <p>Please add this to the CSVs for website rendering</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4022/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (rbharath) [inline examples/tutorials/Introduction_to_Binding_Sites.ipynb:1]: <p>Add some pictures! Much easier for beginners to understanding</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4022/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (rbharath) [inline examples/tutorials/Introduction_to_Binding_Sites.ipynb:1]: <p>Add some pictures</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4022/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (rbharath) [inline examples/tutorials/Introduction_to_Binding_Sites.ipynb:1]: <p>Pictures below please</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4022/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (rbharath) [inline examples/tutorials/Introduction_to_Binding_Sites.ipynb:1]: <p>Maybe make this a bullet list. Check wikipedia for relevant diagrams</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4022/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (rbharath) [inline examples/tutorials/Introduction_to_Binding_Sites.ipynb:1]: <p>Make these links to the docs (the classes to the docs)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4022/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Comment (rbharath) [inline examples/tutorials/Introduction_to_Binding_Sites.ipynb:1]: <p>Maybe split this into 2 tutorials. One basic science, one for PLMs</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4022/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Comment (rbharath) [inline examples/tutorials/Introduction_to_Binding_Sites.ipynb:1]: <p>Add a diagram (look at ESM-2 paper for any architecture diagrams)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4022/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (elisagdelope) [reply examples/tutorials/Introduction_to_Binding_Sites.ipynb:1]: <p>I am removing this section, as suggested to move PLMs (ESM2) to a different, dedicated tutorial (?)</p>

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**